wit/ephemeral/http-types.wit

```diff
@@ -55,4 +55,5 @@ enum http-error {
     invalid-url,
     request-error,
     runtime-error,
+    too-many-requests,
```

Comment:
Just a reminder that this _can be_ a breaking change. I don't see us doing explicit matching on these anywhere in our repos tho, so probably fine.

Aside: it would nice if wit-bindgen had a "non-exhaustive" annotation to emit e.g. Rust's `#[non_exhaustive]`